### ISSUE TITLE
BUG - 2 - All - Accessibility: Screen reader reads Filter and Sort list as 'links' instead of checkbox options

### ISSUE THREAD

**Issue by SKMOstudio** [department-of-veterans-affairs/va-mobile-app#12026]
## What happened?

1. In the Filter and Sort menu on the Travel Claim History page, the screen reader reads out the focused list item as a 'link' instead of a 'checkbox' option that is _checked_ or _not checked_
2. Specifically on Android, the checkbox associated with each list item is a separate focusable object.

## Specs:

- Device: Samsung Galaxy S24, iPhone 15 plus
- OS: Android / iOS
- User Account: Demo
- Accessibility State: Screen reader ON, default settings.

## Steps to Reproduce

1. Log into the app as the Demo user.
2. Go to the Health > Travel Pay page.
3. Tap the Filter and Sort button
4. Focus on a item from Filter menu. 
5. Observe - The screen reader reads out each menu item as a 'link' and does not indicate whether it is 'checked' or 'not checked'.

## Acceptance Criteria

1. When a user focuses on a option in a menu/list that contains a checkbox or radial button, the screen reader should announce the type of object (i.e. "checkbox") and whether it is "checked" or "not checked".
2. Screen readers should focus on the name of the menu/list option and associated checkbox as a single object. The user should not have to do additional navigation to find the checkbox for an option.

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: **High** Low 
- Frequency: High **Low** 

- 3 - Low
- 2 - High
- 1 - Critical

## Screen shot(s) and additional information

https://github.com/user-attachments/assets/ddac49f7-a521-42d7-aac3-eb9b39f89c7b

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the relevant team's bug epic ([Global](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8302), [Health & Benefits](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8303), [API](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8304), [QART](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8305))
